src/toil_lib/tools/preprocessing.py

```diff
@@ -82,221 +84,425 @@ def run_samtools_index(job, bam_id):
+                outputs={'output.bam': None})
+    return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'output.bam'))
+
+
 def run_picard_create_sequence_dictionary(job, ref_id):
     """
-    Use Picard-tools to create reference dictionary
+    Uses Picard to create reference sequence dictionary
 
     :param JobFunctionWrappingJob job: passed automatically by Toil
-    :param str ref_id: FileStoreID for the reference genome
-    :return: FileStoreID for reference dictionary
+    :param str ref_id: FileStoreID for the reference genome fasta file
+    :return: FileStoreID for sequence dictionary file
     :rtype: str
     """
     job.fileStore.logToMaster('Created reference dictionary')
     work_dir = job.fileStore.getLocalTempDir()
     job.fileStore.readGlobalFile(ref_id, os.path.join(work_dir, 'ref.fasta'))
     command = ['CreateSequenceDictionary', 'R=ref.fasta', 'O=ref.dict']
-    docker_call(work_dir=work_dir, parameters=command,
+    docker_call(work_dir=work_dir,
+                parameters=command,
                 tool='quay.io/ucsc_cgl/picardtools:1.95--dd5ac549b95eb3e5d166a5e310417ef13651994e')
     return job.fileStore.writeGlobalFile(os.path.join(work_dir, 'ref.dict'))
 
 
-def run_gatk_preprocessing(job, bam, bai, ref, ref_dict, fai, phase, mills, dbsnp, mem='10G', unsafe=False):
+def picard_mark_duplicates(job, bam, bai):
     """
-    Convenience method for grouping together GATK preprocessing
+    Runs Picard MarkDuplicates on a BAM file. Assumes the BAM file is coordinate sorted.
 
     :param JobFunctionWrappingJob job: passed automatically by Toil
-    :param str bam: Sample BAM FileStoreID
-    :param str bai: Bam Index FileStoreID
-    :param str ref: Reference genome FileStoreID
-    :param str ref_dict: Reference dictionary FileStoreID
-    :param str fai: Reference index FileStoreID
-    :param str phase: Phase VCF FileStoreID
-    :param str mills: Mills VCF FileStoreID
-    :param str dbsnp: DBSNP VCF FileStoreID
-    :param str mem: Memory value to be passed to children. Needed for CI tests
-    :param bool unsafe: If True, runs gatk UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
-    :return: BAM and BAI FileStoreIDs from Print Reads
+    :param bam: FileStoreID for BAM file
+    :param bai: FileStoreID for BAM index file
+    :return: FileStoreIDs for BAM and BAI files
+    :rtype: tuple
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+
+    # Retrieve file path
+    job.fileStore.readGlobalFile(bam, os.path.join(work_dir, 'sorted.bam'))
+    job.fileStore.readGlobalFile(bai, os.path.join(work_dir, 'sorted.bai'))
+
+    # Call: picardtools
+    command = ['MarkDuplicates',
+               'INPUT=sorted.bam',
+               'OUTPUT=mkdups.bam',
+               'METRICS_FILE=metrics.txt',
+               'ASSUME_SORTED=true',
+               'CREATE_INDEX=true',
+               'VALIDATION_STRINGENCY=LENIENT']  # Ignores minor formatting issues
```

Comment:
I like to make the `VALIDATION_STRINGENCY` settings a flag that the user can pass in (with `LENIENT` as the default).